src/test/ui/lint/must_use-tuple.rs

```diff
@@ -0,0 +1,15 @@
+#![deny(unused_must_use)]
+
+fn foo() -> (Result<(), ()>, ()) {
+    (Ok::<(), ()>(()), ())
+}
+
+fn main() {
+    (Ok::<(), ()>(()),); //~ ERROR unused `std::result::Result`
```

Comment:
Can you also add a test that it works in nested tuples? (e.g. `((Err(()), ()), ())`)

The code looks correct to me, but it'd be good to make sure this is tested and the diagnostics don't regress.